src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php

```diff
@@ -166,38 +265,47 @@ public function prepare_item_for_response( $theme, $request ) {
 			}
 		}
 
+		$current_theme = wp_get_theme();
+		if ( rest_is_field_included( 'status', $fields ) ) {
+			$data['status'] = ( $this->is_same_theme( $theme, $current_theme ) ) ? 'active' : 'inactive';
+		}
+
 		if ( rest_is_field_included( 'theme_supports', $fields ) ) {
-			foreach ( get_registered_theme_features() as $feature => $config ) {
-				if ( ! is_array( $config['show_in_rest'] ) ) {
-					continue;
-				}
+			if ( $this->is_same_theme( $theme, $current_theme ) ) {
```

Comment:
Let's make this bail early instead if it isn't the same theme. That way we don't have to add another layer of nesting.

Reply:
Pseduo code
```
if ( ! is_same_theme ) {
    $theme_supports = [];
    continue;
}
```

Reply:
There is an else here https://github.com/WordPress/wordpress-develop/pull/264/files#diff-66202903461c5842c77ef3e6d8203a1e869efda4c633782c59cb189f04f4350dR308

Reply:
Actually using `continue` isn't an option here as this logic is outside a loop. We could make code more readable by repeating ourselves a little like

```php
if ( rest_field_included(....) && ! $is_same_theme) {
    $theme_supports = [];
} elseif ( rest_field_included(....) && $is_same_theme) {
 .....
}
```

Basically above is the same from logic perspective but without nested ifs. What are your thoughts @spacedmonkey and @TimothyBJacobs ?

Reply:
I like the logic how it is.

Reply:
> Actually using continue isn't an option here as this logic is outside a loop.

Ah my mistake.

Do we need to populate `theme_supports` at all if it isn't the current theme? I think we should just omit it, and in that case we can simplify the conditional to `is field included && is_same_theme`.

Reply:
@TimothyBJacobs As long as you all are comfortable with inconsistent responses for theme structure - only active one will contain `theme_supports` and others would not have it. If that's not an issue, and this entire field can be optional then I'll remove populating it for non active themes.

Reply:
I think it works. Active themes don't have any theme supports, so omitting the field makes sense. Ideally, if we caught this issue before 5.0, we would've probably used `context` to differentiate the available fields on the active theme vs all themes. And when a field isn't available in a given `context` we remove it.

Reply:
Thanks @TimothyBJacobs . I am making that change now.

Reply:
@TimothyBJacobs this change has been made.